libsolidity/CMakeLists.txt

```diff
@@ -127,6 +127,8 @@ set(sources
 	interface/CompilerStack.cpp
 	interface/CompilerStack.h
 	interface/DebugSettings.h
+	interface/FileReader.cpp
```

Comment:
Why have this in interface? Since lsp is integrated into solc, cannot it just use the same ReadFile interface and keeping the filereader implementation in solc?

Reply:
I'm actually doing it the other way, since there is quite some logic shared between solc file reading and solc-lsp file reading, I've encapsulated that into the FileReader class. This change is all in one commit and could be extracted into its own smaller PR).

Reply:
If there's FileReader in the interface, then why do we need ReadFile? It just looks confusing and complicated. All that file reading via boost does not look like an interface, rather a heavy implementation.

Can you explain the flow and why this is needed? Is it possible to untangle lsp?

Reply:
I think ReadFile can then go. Why it is needed, hm, I felt like I am duplicating quite some code that solc naturally is doing too, for gathering files for compilations. I did not want to duplicate all these lines, and was instead creating the FileReader to be shared between the two (solc CLI and solc LSP). I would have to see if we use the type `ReadFile` somewhere else, too, and if not, I think I'd remove that and directly pass `FileReader` as part of the specific commit.

Reply:
What I don't get is why does FileReader needs to be part of the lsp internals and why can't it just access files through ReadFile (like the compiler is doing)?